cmd/cli/serve/serve.go

```diff
@@ -517,6 +503,25 @@ func buildEnvVariables(ctx context.Context, nodeConfig *node.NodeConfig, ipfsCon
 	return envVarBuilder.String(), nil
 }
 
+func getPublicNATSOrchestratorURL(nodeConfig *node.NodeConfig) *url.URL {
+	orchestrator := &url.URL{
+		Scheme: "nats",
+		Host:   nodeConfig.NetworkConfig.AdvertisedAddress,
+	}
+
+	// Only display the secret if the user did not set it explicitly.
+	// Else, they should already know it!
+	if secret, err := config.Get[string](types.NodeNetworkAuthSecret); err == nil && secret == "" {
+		orchestrator.User = url.User(nodeConfig.NetworkConfig.AuthSecret)
```

Comment:
If `nodeConfig.NetworkConfig.AuthSecret` is "" this eventually generates a URL like `://@host:port/`. While this is entirely valid,  it seems like it might be a cause of support requests on unrelated issues.  Perhaps if we don't have a user to set in the URL we should just not set it?

Reply:
Hm. If `nodeConfig.NetworkConfig.AuthSecret` starts out at `nil` then this code will make sure it is filled in?
https://github.com/bacalhau-project/bacalhau/blob/3920ce72201feddf0e21cc6856e9a4028df4d0ad/cmd/cli/serve/util.go#L198

But fair enough, we may as well be defensive and check the value here too.